### ISSUE TITLE
[Bug]: CE ignore weapon restrictions of mod (VWE-Heavy & VFE-Pirates)

### ISSUE THREAD

**Issue by skycrossercat** [CombatExtended-Continued/CombatExtended#2726]
Rimworld version: 1.4.3704
Combat Extended version: 1.4.5.4
Combat Extended source: latest dev snapshot downloaded from discord pin message
Your operating system: Windows 10
Your mod list: https://rentry.co/v3vi4r

**Description**
VWE-Heavy and VFE-Pirates both add weapon restrictions via `<modExtension>` to their own weapons' ThingDef, which behaves normally in vanilla, but in CE, it's ignored totally, every pawn can equip the restricted weapons

(if having wrong load order for VFE-Pirates, the CE patches wouldn't apply, then the restrictions return to vanilla behavior, so maybe smth overwrite it in the `"CombatExtended.PatchOperationMakeGunCECompatible"` process, or make them CE weapons stop Vanilla Expanded Framework to recognize them)

**Expected behavior**
Pawns should follow the restrictions and not able to equip the weapons unless they fulfill the requirement

**To reproduce**
Have VWE-Heavy or VFE-Pirates in mod list while CE is loaded, spawn weapons from the mods, order pawns to equip those weapons

**Screenshots & log dumps**
**in CE**
![temp1](https://github.com/CombatExtended-Continued/CombatExtended/assets/58024063/7ff60bea-e2a5-481a-8315-677dd71c0e6a)
all testing pawns are normal baseliners without required traits or gears
![temp2](https://github.com/CombatExtended-Continued/CombatExtended/assets/58024063/8b7fb507-8329-4b23-b9f3-9ec402989424)
in CE enviorment, they can equip weapons from VWE-Heavy
![temp3](https://github.com/CombatExtended-Continued/CombatExtended/assets/58024063/f557b210-e4cd-4a2b-bfa5-88c1c8fe8ebe)
and weapons from VFE-Pirates (warcaskets weapon

**in Vanilla**
![temp5](https://github.com/CombatExtended-Continued/CombatExtended/assets/58024063/2cb3ae18-a0b0-4f90-b7b1-b0ca2994b2e5)
![temp6](https://github.com/CombatExtended-Continued/CombatExtended/assets/58024063/5656d694-f206-4a9d-b9d7-8773a673baec)
normally, they should have such restrictions like in vanilla

Logs: https://gist.github.com/HugsLibRecordKeeper/f4375cc438ba5dcfe73bdd764f9fa5b3
(although there's no errors while testing, still provide this just in case
**Complete the following checklist**
I hereby verify that I have done the following:
- [x] Confirmed that my game version and load order are correct.
- [x] Confirmed that I am running the appropriate and most updated version of Combat Extended and required compatibility patches.
- [x] Confirmed I am not running any mods with known incompatibilities with Combat Extended.
- [x] Disabled Combat Extended and attempted to reproduce the behavior without success.

**Comment by FayFlightee** [comment]
Frankly I thought this was intended.
It's offset by TERRIBLE weight and bulk, which means you'd need an exoskeleton or something to wield it anyways.

**Comment by skycrossercat** [comment]
Not really, your pawn can still grab it even though they don't have enough capacity. Those guns in my screenshots are at least 50kg while standard pawn only has 40kg carry weight, they may not able to carry extra ammunition, but they can still fire it if there's ammo already loaded.
Also, in the original patch thread #439, they already knew the issue, but without a way to fix that, they kept it the way you mentioned, it's more like a temporary work around.

**Comment by JulekJulas** [comment]
This is a big issue when raiding pirate cities as I seen many completely naked pawns with just warcasket weapons with no ammo because the weapons took all of their weight.
